packages/buidler-core/src/builtin-tasks/node.ts

```diff
@@ -63,6 +72,66 @@ Private Key: ${privateKey}
 }
 
 export default function () {
+  internalTask(TASK_NODE_WATCH_COMPILER_OUTPUT)
+    .addParam("url", "The URL of the node", undefined, types.string)
+    .setAction(async ({ url }: { url: string }, { config }) => {
+      const { default: fetch } = await import("node-fetch");
+      const chokidar = await import("chokidar");
+
+      const compilerVersion = config.solc.version;
+      const solcInputPath = path.join(config.paths.cache, SOLC_INPUT_FILENAME);
+      const solcOutputPath = path.join(
+        config.paths.cache,
+        SOLC_OUTPUT_FILENAME
+      );
+
+      const addCompilationResult = async () => {
+        if (
+          !(await fsExtra.pathExists(
+            path.join(config.paths.cache, SOLC_INPUT_FILENAME)
+          ))
+        ) {
+          return false;
+        }
+
+        if (
+          !(await fsExtra.pathExists(
+            path.join(config.paths.cache, SOLC_OUTPUT_FILENAME)
+          ))
+        ) {
+          return false;
+        }
+
+        const compilerInput = await fsExtra.readJSON(solcInputPath, {
+          encoding: "utf8",
+        });
+        const compilerOutput = await fsExtra.readJSON(solcOutputPath, {
+          encoding: "utf8",
+        });
+
+        await fetch(url, {
```

Comment:
I think this adds some complexity that we may not need, as we can use the provider directly. 

Maybe we need to move this to a function instead of a task to do that.

Reply:
I'm not sure if I understood this. You mean replacing the `TASK_NODE_WATCH_COMPILER_OUTPUT` task with a function, so that we can get the provider from the `JsonRpcServer` and pass it to that function?

Reply:
yep